digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/strategy/TrackCalculatorStrategy.scala

```diff
@@ -167,6 +167,8 @@ trait TrackCalculatorStrategy {
 
   protected def adjustTwoTracks(startAddress: Option[Long], leftProjectLinks: Seq[ProjectLink], rightProjectLinks: Seq[ProjectLink], calibrationPoints: Map[Long, UserDefinedCalibrationPoint],
                                 restLeftProjectLinks: Seq[ProjectLink] = Seq(), restRightProjectLinks: Seq[ProjectLink] = Seq()): TrackCalculatorResult = {
+    if (leftProjectLinks.isEmpty || rightProjectLinks.isEmpty)
```

Comment:
Should the exception be `MissingTrackException`? And the error message: "Missing left track." or "Missing right track"?

Reply:
Since we never know which track is missing, that will imply writing unnecessary if/else or pattern matching statement. The more generic message were already being used, so i took it also for this case. I'm also logging the size of both tracks, in case it's needed for debugging in the future. We can always check which track is actually missing.

Reply:
In any case I think the exception should be `MissingTrackException`, because this is not about different roadway count on each track but other track completely missing.